src/katgpucbf/fgpu/engine.py

```diff
@@ -494,7 +495,7 @@ def __init__(self, output: Output, engine: "Engine", context: AbstractContext, d
         if isinstance(output, NarrowbandOutput):
             narrowband_config = NarrowbandConfig(
                 decimation=output.decimation,
-                mix_frequency=-output.centre_frequency / engine.adc_sample_rate,
+                mix_frequency=-Fraction(output.centre_frequency) / Fraction(engine.adc_sample_rate),
```

Comment:
Aha, and doing `Fraction(centre_frequency / adc_sample_rate)` would result in the exact thing you're trying to avoid by introducing this update.